client/components/date-range/index.js

```diff
@@ -0,0 +1,300 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { bindAll, noop } from 'lodash';
+import { DateUtils } from 'react-day-picker';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { localize } from 'i18n-calypso';
+import Gridicon from 'gridicons';
+import DatePicker from 'components/date-picker';
+import Button from 'components/button';
+import Popover from 'components/popover';
+import DateRangeInputs from './inputs';
+import DateRangeHeader from './header';
+
+export class DateRange extends Component {
+	static propTypes = {
+		onDateSelect: PropTypes.func,
+		onDateCommit: PropTypes.func,
+	};
+
+	static defaultProps = {
+		onDateSelect: noop,
+		onDateCommit: noop,
+	};
+
+	static dateFormat = 'MM/DD/YYYY';
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			popoverVisible: false,
+			oldStartDate: '',
+			oldEndDate: '',
+			startDate: this.props.startDate || this.props.moment().subtract( 1, 'months' ),
+			endDate: this.props.endDate || this.props.moment(),
+			oldDatesSaved: false,
+			inputFromDate:
+				this.dateToHumanReadable( this.props.startDate ) ||
+				this.dateToHumanReadable( this.props.moment().subtract( 1, 'months' ) ),
+			inputToDate:
+				this.dateToHumanReadable( this.props.endDate ) ||
+				this.dateToHumanReadable( this.props.moment() ),
+		};
+
+		bindAll( this, [
+			'togglePopover',
+			'closePopover',
+			'onSelectDate',
+			'revertDates',
+			'commitDates',
+			'handleInputChange',
+			'handleInputBlur',
+		] );
+
+		this.startButtonRef = React.createRef();
+	}
+
+	togglePopover() {
+		this.setState( {
+			popoverVisible: ! this.state.popoverVisible,
+			inputFromDate: this.dateToHumanReadable( this.state.startDate ),
+			inputToDate: this.dateToHumanReadable( this.state.endDate ),
+		} );
+
+		// Close the popover
+		this.revertDates();
+	}
+
+	closePopover() {
+		this.setState( {
+			popoverVisible: false,
+		} );
+	}
+
+	handleInputChange( e ) {
+		const val = e.target.value;
+
+		const fromOrTo = e.target.id.includes( 'start' ) ? 'From' : 'To';
+
+		this.setState( {
+			[ `input${ fromOrTo }Date` ]: val,
+		} );
+	}
+
+	handleInputBlur( e ) {
+		const val = e.target.value;
+		const startOrEnd = e.target.id;
```

Comment:
This worries me for similar reasons - we're relying on something quite implicit. I think we'd be able to do something similar to the suggestion above.

Reply:
👍

Reply:
See above